### PR TITLE
Move fallible ti.task.dag assignment back inside try/except block (#24533)

### PR DESCRIPTION
**Pull request by ejk43**
It looks like ti.task.dag was originally protected inside try/except, but was moved out at commit 7be87d

This causes the "View Logs in Elasticsearch" option to crash in the Airflow UI

related: #24533

### CODE REVIEW

File: airflow/providers/elasticsearch/log/es_task_handler.py
Comment (uranusjr) [inline airflow/providers/elasticsearch/log/es_task_handler.py]: Judging from the comment on the `except` clause, this should be

```python
try:
    dag = ti.task.dag
except AttributeError:  # ti.task is not always set.
    data_interval: Tuple[datetime, datetime] = (dag_run.data_interval_start, dag_run.data_interval_end)
else:
    assert dag is not None  # For Mypy.
    data_interval = dag.get_run_data_interval(dag_run)
```

instead.

Comment (ejk43) [reply airflow/providers/elasticsearch/log/es_task_handler.py]: Fair request -- updated

File: airflow/providers/elasticsearch/log/es_task_handler.py
Comment (uranusjr) [inline airflow/providers/elasticsearch/log/es_task_handler.py]:
```suggestion
            data_interval = dag.get_run_data_interval(dag_run)
```

I _think_ this can be removed now?

File: airflow/utils/log/file_task_handler.py
Comment (uranusjr) [inline airflow/utils/log/file_task_handler.py]:
```suggestion
                data_interval = dag.get_run_data_interval(dag_run)
```

**Review by uranusjr — APPROVED** [review]

**Comment by ejk43** [comment]
Another good catch. Thanks for the review and commit, I appreciate the detail.

What's the process to get this into main branch from here? Is anything else needed? (first time PR for me)

**Comment by uranusjr** [comment]
I’m just going to merge this even without a second review since I wrote the original example.

**Comment by uranusjr** [comment]
We generally need two committer reviews to merge a PR, so let’s wait for now.

**Pull request merged**
